### DISCUSSION TITLE
ArgoCD as part of a PullRequest-Review process?

### DISCUSSION THREAD

**Discussion by meysholdt** [argoproj/argo-cd#7181] [Q&A]
Hallo!

While I'm a big fan of ArgoCD, there is a bit of struggle to integrate it into our processes:

Can ArgoCD be used to show App Diff between the live app and the desired state from a *feature* branch? 
I know ArgoCD is excellent at showing that diff between the life app and the tracked git branch. However, I would love to see the diff for changes *before* they get merged onto the tracked branch. For example, it would be valuable to see such a diff as part of a PullRequest-Review.

I'm aware of https://github.com/argoproj-labs/argocd-bot, but it seems [unmaintained for +2 years now](https://github.com/argoproj-labs/argocd-bot/issues/34).

I'm aware that `argocd app diff --revision <featurebranch>` can create the diff that I'm asking for... but is there a solution that has a WebUI and maybe integrates with PullRequests?

Can the ArgoCD WebUI show App Diffs for git-branches without actually changing the app manifest? For example can ArgoCD render an App Diff for a URL such as `https://<argo-cd>/applications/<appname>?tab=diff&revision=<myfeaturebranch>`. Note the `revision=<myfeaturebranch>` part.

Do others out here use ArgoCD-provided App Diffs as part of a PullRequest-review? if so, how? If not, why?

**Comment by pacoguzman** [comment]
Hi @meysholdt , do you find an answer for this? I'm stuck in the same thing to integrate ArgoCD in our review process. Your question is similar to https://github.com/argoproj/argo-cd/discussions/7524 as not only see the diff but make able to validate/lint the changes before merging them and apply onto the clusters.

By the way, when you mention `argocd app diff --revision <featurebranch>` what are the changes that need to be included on the `<featurebranch>` the Application definition? or the Application and the Source definition? (we're using a mono repo which includes Application and Source definition (using kustomize) and want to validate the Source with some custom policies before merging PR too)

Thanks in advance

**Reply by colethienes** [reply]
Hi @meysholdt, I couldn't find a solution to this for my company, so I built and open-sourced our solution. See [Infro](https://github.com/infro-io/infro-core).
